Source/PINDiskCache.m

```diff
@@ -673,7 +697,7 @@ - (void)trimDiskToDate:(NSDate *)trimDate
         
         for (NSString *key in keysSortedByDate) { // oldest files first
             NSDate *accessDate = _metadata[key].date;
-            if (!accessDate)
+            if (!accessDate || _metadata[key].ageLimit > 0.0)
```

Comment:
Yeah, I agree it is sort of confusing. My thoughts here are that the `trimDate` that is passed in here is based on the cache-level age limit and we don't want to remove objects that have overridden the age limit.

I don't know what the correct thing to do here would be. -trimToAgeLimitRecursively calling -removeExpiredObjectsAsync would certainly be an option, but I wanted to avoid over-calling it since it has to iterate over all of the objects in the cache to determine if they are expired or not. Although, I don't think there is much harm in letting these expired objects lay dormant on the filesystem until the next initialization; the getters/iterators will treat them as if they don't exist.

Reply:
Ok, that makes sense, I think if we address the issue I mention below regarding `trimDiskToSizeByDate` and `trimToCostLimitByDate`

Reply:
Done.